src/main/java/cyrus/parser/ParseInfo.java

```diff
@@ -0,0 +1,75 @@
+package cyrus.parser;
+
+import java.util.HashMap;
+
+import cyrus.commands.CommandType;
+
+/**
+ * Packet of information about the parsing response.
+ *
+ * <p>Modelled after CLI applications where the {@code argument} is the text before any options
+ * and an {@code option} is the values of "flags".</p>
+ */
+public class ParseInfo {
```

Comment:
I like how you have implemented this task to parse the commands more efficiently! However, perhaps you can consider doing this within Parser class instead?